### ISSUE TITLE
Error: Invalid value for variable

### ISSUE THREAD

**Issue by kevinevans1** [Azure/terraform-azurerm-caf-enterprise-scale#241]
Error: Invalid value for variable
│ 
│   on .terraform/modules/enterprise_scale/main.tf line 12, in module "management_group_archetypes":
│   12:   scope_id                = each.key
│ 
│ The scope_id value must be a valid Subscription or Management Group ID.
│ 
│ This was checked by the validation rule at .terraform/modules/enterprise_scale/modules/archetypes/variables.tf:21,3-13.

**Comment by jtracey93** [comment]
Hey @kevinevans1,

Can you provide some gists and/or code snippets of your config so we can take a look?

**Comment by kevinevans1** [comment]
see attached :) @jtracey93 

`# We strongly recommend using the required_providers block to set the
# Azure Provider source and version being used.

# You can use the azurerm_client_config data resource to dynamically
# extract connection settings from the provider configuration.

data "azurerm_client_config" "core" {}

# Call the caf-enterprise-scale module directly from the Terraform Registry
# pinning to the latest version

module "enterprise_scale" {
  source  = "Azure/caf-enterprise-scale/azurerm"
  version = "1.1.0"

  providers = {
    azurerm              = azurerm
    azurerm.connectivity = azurerm
    azurerm.management   = azurerm
  }

  root_parent_id = data.azurerm_client_config.core.tenant_id
  root_id        = "DC"
  root_name      = "Justice League"

  deploy_management_resources   = true
  subscription_id_management    = "sub string"
  deploy_connectivity_resources = true
  subscription_id_connectivity  = "sub string"
  deploy_identity_resources     = true
  subscription_id_identity      = "sub string"

}`

Where sub string would be , i would populate with the subscription ID

**Comment by krowlandson** [comment]
So in this case, the `scope_id` variable within the `archetype` child module uses the following validation logic:

```hcl
can(regex("^/(subscriptions|providers/Microsoft.Management/managementGroups)/[a-z0-9-]{2,36}$", var.scope_id))
```

By comparison, the `root_id` variable within the root module uses the following validation logic:

```hcl
can(regex("^[a-zA-Z0-9-]{2,10}$", var.root_id))
```

This allows you to define a Management Group ID which contains capitals, and leads to the error above when parsed to the `archetype` child module for processing.

I will raise a PR to fix this for consistent handling.

Additionally, you will also need to add a `provider` block to your configuration if you haven't already.

**Comment by kevinevans1** [comment]
> 

Thanks for highlighting this , to be honest I shouldn't be using capitals in group names 😊  , but the choice is there for the community just in case @krowlandson

**Comment by krowlandson** [comment]
Included in [release `v1.1.1`](https://github.com/Azure/terraform-azurerm-caf-enterprise-scale/releases/tag/v1.1.1)
